plugins/catalog/src/components/EntityContextMenu/EntityContextMenu.tsx

```diff
@@ -84,6 +86,24 @@ export function EntityContextMenu(props: EntityContextMenuProps) {
     setAnchorEl(undefined);
   };
 
+  const [state, setState] = useState({
+    open: false,
+    vertical: 'top',
+    horizontal: 'right',
+  });
+
+  const { open, vertical, horizontal } = state;
+
+  const handleClose = () => {
+    setState({ ...state, open: false });
+  };
```

Comment:
I'll just note that an optimization here is to use the function form of state setters. We don't need to regenerate these functions over and over.

```ts
const handleClose = useCallback(() => {
  setState(s => { ...s, open: false });
}, []); // no dependencies are needed here
```

same with copyToClipboard below.

Reply:
Done, it was needed for better performance.